Reachable Panic in `RTM_GETLINK` handler on non-UTF-8 `IFLA_IFNAME`

### Describe the bug

Asterinas panics when processing a `RTM_GETLINK` request whose `IFLA_IFNAME` attribute contains non-UTF-8 bytes.
The panic is caused by calling `unwrap()` on user-controlled netlink input in

https://github.com/asterinas/asterinas/blob/356dd6e71bc6f980f7ebb883728f2e912553bb12/kernel/src/net/socket/netlink/route/kernel/link.rs#L74

### To Reproduce
Compile and run the PoC
```c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <sys/socket.h>
#include <linux/netlink.h>
#include <linux/rtnetlink.h>

#ifndef IFLA_IFNAME
#define IFLA_IFNAME 3
#endif

struct nlattr_hdr {
    unsigned short nla_len;
    unsigned short nla_type;
};

int main(void) {
    int fd = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
    if (fd < 0) return 1;

    struct sockaddr_nl local = {0};
    local.nl_family = AF_NETLINK;
    local.nl_pid = getpid();
    if (bind(fd, (struct sockaddr *)&local, sizeof(local)) < 0) return 1;

    unsigned char bad_name[] = {0xFF, 0xFE, 0x00};
    unsigned short attr_len = sizeof(struct nlattr_hdr) + sizeof(bad_name);
    unsigned short attr_padded = (attr_len + 3) & ~3;

    unsigned int msg_len = sizeof(struct nlmsghdr) +
                           sizeof(struct ifinfomsg) +
                           attr_padded;

    unsigned char *buf = calloc(1, msg_len);
    if (!buf) return 1;

    struct nlmsghdr *nlh = (struct nlmsghdr *)buf;
    nlh->nlmsg_len = msg_len;
    nlh->nlmsg_type = RTM_GETLINK;
    nlh->nlmsg_flags = NLM_F_REQUEST;
    nlh->nlmsg_seq = 1;
    nlh->nlmsg_pid = getpid();

    struct ifinfomsg *ifi = (struct ifinfomsg *)(buf + sizeof(*nlh));
    ifi->ifi_family = AF_UNSPEC;
    ifi->ifi_index = 0;

    struct nlattr_hdr *nla = (struct nlattr_hdr *)(buf + sizeof(*nlh) + sizeof(*ifi));
    nla->nla_len = attr_len;
    nla->nla_type = IFLA_IFNAME;
    memcpy((char *)nla + sizeof(*nla), bad_name, sizeof(bad_name));

    struct sockaddr_nl kernel = {0};
    kernel.nl_family = AF_NETLINK;

    struct iovec iov = {
        .iov_base = buf,
        .iov_len = msg_len,
    };

    struct msghdr msg = {
        .msg_name = &kernel,
        .msg_namelen = sizeof(kernel),
        .msg_iov = &iov,
        .msg_iovlen = 1,
    };

    sendmsg(fd, &msg, 0);

    free(buf);
    close(fd);
    return 0;
}
```

### Expected behavior
In Linux, the PoC file returns with `ENODEV` error.

### Environment

- Official docker asterinas/asterinas:0.17.1-20260319
- Asterinas version: main 356dd6e71bc6f980f7ebb883728f2e912553bb12

### Logs
```
~ # /etc/ex1927.c.bin 
[    97.295] ERROR : Uncaught panic:
        called `Result::unwrap()` on an `Err` value: Utf8Error { valid_up_to: 0, error_len: Some(1) }
        at /root/asterinas/kernel/src/net/socket/netlink/route/kernel/link.rs:74
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, is_exited: false, cpu_affinity: AtomicIdSet { bits: [1], phantom: PhantomData<ostd::cpu::id::CpuId> }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 34529300 }, fair: FairAttr { weight: 1024, pending_weight: UnsafeCell { .. }, vruntime: 542026646 } } })
make: *** [Makefile:273: run_kernel] Error 1
```

--- a/kernel/libs/aster-bigtcp/src/iface/common.rs
+++ b/kernel/libs/aster-bigtcp/src/iface/common.rs
@@ -2,11 +2,12 @@
 
 use alloc::{
     collections::btree_map::{BTreeMap, Entry},
-    string::String,
+    ffi::CString,
     sync::Arc,
     vec::Vec,
 };
 use core::{
+    ffi::CStr,
     ops::Deref,
     sync::atomic::{AtomicBool, AtomicU32, Ordering},
 };
@@ -37,7 +38,7 @@ use crate::{
 
 pub struct IfaceCommon<E: Ext> {
     index: u32,
-    name: String,
+    name: CString,
     type_: InterfaceType,
     flags: InterfaceFlags,
 
@@ -81,7 +82,7 @@ impl From<IpAddress> for NormalizedAddress {
 
 impl<E: Ext> IfaceCommon<E> {
     pub(super) fn new(
-        name: String,
+        name: CString,
         type_: InterfaceType,
         flags: InterfaceFlags,
         interface: smoltcp::iface::Interface,
@@ -105,7 +106,7 @@ impl<E: Ext> IfaceCommon<E> {
         self.index
     }
 
-    pub(super) fn name(&self) -> &str {
+    pub(super) fn name(&self) -> &CStr {
         &self.name
     }
 

--- a/kernel/libs/aster-bigtcp/src/iface/iface.rs
+++ b/kernel/libs/aster-bigtcp/src/iface/iface.rs
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: MPL-2.0
 
 use alloc::sync::Arc;
+use core::ffi::CStr;
 
 use smoltcp::wire::{Ipv4Address, Ipv4Cidr, Ipv6Address};
 
@@ -58,7 +59,7 @@ impl<E: Ext> dyn Iface<E> {
     /// Gets the name of the iface.
     ///
     /// In Linux, the name is usually the driver name followed by a unit number.
-    pub fn name(&self) -> &str {
+    pub fn name(&self) -> &CStr {
         self.common().name()
     }
 

--- a/kernel/libs/aster-bigtcp/src/iface/phy/ether.rs
+++ b/kernel/libs/aster-bigtcp/src/iface/phy/ether.rs
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use alloc::{collections::btree_map::BTreeMap, string::String, sync::Arc};
+use alloc::{collections::btree_map::BTreeMap, ffi::CString, sync::Arc};
 
 use aster_softirq::BottomHalfDisabled;
 use ostd::sync::SpinLock;
@@ -37,7 +37,7 @@ impl<D: WithDevice, E: Ext> EtherIface<D, E> {
         ether_addr: EthernetAddress,
         ip_cidr: Ipv4Cidr,
         gateway: Ipv4Address,
-        name: String,
+        name: CString,
         sched_poll: E::ScheduleNextPoll,
         flags: InterfaceFlags,
     ) -> Arc<Self> {

--- a/kernel/libs/aster-bigtcp/src/iface/phy/ip.rs
+++ b/kernel/libs/aster-bigtcp/src/iface/phy/ip.rs
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use alloc::{string::String, sync::Arc};
+use alloc::{ffi::CString, sync::Arc};
 
 use smoltcp::{
     iface::Config,
@@ -30,7 +30,7 @@ impl<D: WithDevice, E: Ext> IpIface<D, E> {
         driver: D,
         ip_cidr: Ipv4Cidr,
         ipv6_cidr: Option<Ipv6Cidr>,
-        name: String,
+        name: CString,
         sched_poll: E::ScheduleNextPoll,
         type_: InterfaceType,
         flags: InterfaceFlags,

--- a/kernel/src/net/iface/init.rs
+++ b/kernel/src/net/iface/init.rs
@@ -1,6 +1,5 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use alloc::borrow::ToOwned;
 use core::slice::Iter;
 
 use aster_bigtcp::{
@@ -99,7 +98,7 @@ fn new_loopback() -> Arc<Iface> {
             LOOPBACK_IPV6_ADDRESS,
             LOOPBACK_IPV6_PREFIX_LEN,
         )),
-        "lo".to_owned(),
+        CString::new("lo").unwrap(),
         PollScheduler::new(),
         InterfaceType::LOOPBACK,
         flags,
@@ -148,7 +147,7 @@ fn new_virtio() -> Option<Arc<Iface>> {
         EthernetAddress(ether_addr),
         Ipv4Cidr::new(VIRTIO_ADDRESS, VIRTIO_ADDRESS_PREFIX_LEN),
         VIRTIO_GATEWAY,
-        "eth0".to_owned(),
+        CString::new("eth0").unwrap(),
         PollScheduler::new(),
         flags,
     ))

--- a/kernel/src/net/iface/poll.rs
+++ b/kernel/src/net/iface/poll.rs
@@ -26,7 +26,7 @@ pub(super) fn poll_ifaces() {
 
 fn spawn_background_poll_thread(iface: Arc<Iface>) {
     let task_fn = move || {
-        debug!("spawn background poll thread for {}", iface.name());
+        debug!("spawn background poll thread for {:?}", iface.name());
 
         let sched_poll = iface.sched_poll();
         let wait_queue = sched_poll.polling_wait_queue();

--- a/kernel/src/net/socket/netlink/route/kernel/addr.rs
+++ b/kernel/src/net/socket/netlink/route/kernel/addr.rs
@@ -2,6 +2,7 @@
 
 //! Handle address-related requests.
 
+use alloc::borrow::ToOwned;
 use core::num::NonZeroU32;
 
 use super::util::finish_response;
@@ -60,7 +61,7 @@ fn iface_to_new_addr(request_header: &CMsgSegHdr, iface: &Arc<Iface>) -> Option<
 
     let attrs = vec![
         AddrAttr::Address(ipv4_addr.octets()),
-        AddrAttr::Label(CString::new(iface.name()).unwrap()),
+        AddrAttr::Label(iface.name().to_owned()),
         AddrAttr::Local(ipv4_addr.octets()),
     ];
 

--- a/kernel/src/net/socket/netlink/route/kernel/link.rs
+++ b/kernel/src/net/socket/netlink/route/kernel/link.rs
@@ -2,6 +2,7 @@
 
 //! Handle link-related requests.
 
+use alloc::borrow::ToOwned;
 use core::num::NonZero;
 
 use aster_bigtcp::iface::InterfaceType;
@@ -46,7 +47,7 @@ pub(super) fn do_get_link(request_segment: &LinkSegment) -> Result<Vec<RtnlSegme
 
 enum FilterBy<'a> {
     Index(u32),
-    Name(&'a str),
+    Name(&'a CStr),
     Dump,
 }
 
@@ -71,7 +72,7 @@ impl<'a> FilterBy<'a> {
 
         let required_name = request_segment.attrs().iter().find_map(|attr| {
             if let LinkAttr::Name(name) = attr {
-                Some(name.to_str().unwrap())
+                Some(name.as_c_str())
             } else {
                 None
             }
@@ -137,7 +138,7 @@ fn iface_to_new_link(request_header: &CMsgSegHdr, iface: &Arc<Iface>) -> LinkSeg
     };
 
     let attrs = vec![
-        LinkAttr::Name(CString::new(iface.name()).unwrap()),
+        LinkAttr::Name(iface.name().to_owned()),
         LinkAttr::Mtu(iface.mtu() as u32),
     ];
 

--- a/test/initramfs/src/regression/network/netlink_route.c
+++ b/test/initramfs/src/regression/network/netlink_route.c
@@ -2,6 +2,7 @@
 
 #include <net/if.h>
 #include <netlink/route/addr.h>
+#include <stddef.h>
 #include <unistd.h>
 
 #include "../common/test.h"
@@ -215,6 +216,14 @@ FN_TEST(get_link_error)
 
 	// ifname = "" with nul
 	req.abuf[0] = '\0';
+	TEST_ERROR_SEGMENT(ENODEV);
+
+	// ifname = "\xff\xfe" with nul
+	const unsigned char non_utf8_ifname[] = { 0xff, 0xfe, '\0' };
+	req.hdr.nlmsg_len =
+		offsetof(struct nl_req, abuf) + sizeof(non_utf8_ifname);
+	req.ahdr.nla_len = sizeof(struct nlattr) + sizeof(non_utf8_ifname);
+	memcpy(req.abuf, non_utf8_ifname, sizeof(non_utf8_ifname));
 	TEST_ERROR_SEGMENT(ENODEV);
 
 	// Invalid name attribute (too short) without index

